otherlibs/systhreads/st_stubs.c

```diff
@@ -215,10 +217,23 @@ static void caml_thread_leave_blocking_section(void)
   /* Update Current_thread to point to the thread descriptor corresponding to
      the thread currently executing */
   Current_thread = st_tls_get(Thread_key);
+  atomic_store_explicit(&Caml_state->lock_owner, (uintnat)Current_thread,
+                        memory_order_relaxed);
   /* Restore the runtime state from the curr_thread descriptor */
   caml_thread_restore_runtime_state();
 }
 
+static int thread_has_lock(void)
```

Comment:
Is there a reason this isn't consistent with the other hooks' default implementations and prefixed with `caml_`?

Reply:
I did not realise there was such a convention, to this was just a static function, so without a prefix being needed.